### PR TITLE
Fix/windows newlines

### LINKED ISSUE #35: Fix issues reading files which include CRLF
Modelica-builder does not work for files that use CRLF (`\r\n`) instead of only newlines (`\n`). Modelica-builder should work for either line delimiter.

### PR DESCRIPTION
**Pull request by macintoshpie**
## changes
- fix handling of files created on Windows, which use \r\n for line separators instead of \n

## issues
Closes #35

### CODE REVIEW

File: tests/test_builder.py [hunk 1]
```diff
@@ -43,15 +46,15 @@ def test_component_builder_at_index_0(self):
         with open(os.path.join(self.output_dir, 'DCMotor_component_builder_0.mo'), 'w') as file:
             file.write(result)
 
-        with open(os.path.join(self.data_dir, 'DCMotor.mo'), 'r') as f:
+        with open(os.path.join(self.data_dir, test_file), 'r') as f:
             diffs = get_diffs(f.readlines(), result.splitlines(keepends=True))
 
-            self.assertEqual(1, len(diffs['additions']), f'should have 1 addition, has these: {diffs["additions"]}')
-            self.assertEqual(0, len(diffs['deletions']), f'should have 0 deletions, has these: {diffs["deletions"]}')
+            assert 1 == len(diffs['additions']), f'should have 1 addition, has these: {diffs["additions"]}'
```
Comment (macintoshpie) [inline tests/test_builder.py:52]: no longer inheriting from unittest TestCase so that we can parametrize the class. Move to pytest method of assertions

Comment (nllong) [reply tests/test_builder.py:52]: so sad, I like the `self.assert*` methods :(

File: tests/test_builder.py [hunk 2]
```diff
@@ -97,10 +100,10 @@ def test_component_builder_at_end(self):
         with open(os.path.join(self.output_dir, 'DCMotor_component_builder_at_end.mo'), 'w') as file:
             file.write(result)
 
-        with open(os.path.join(self.data_dir, 'DCMotor.mo'), 'r') as f:
+        with open(os.path.join(self.data_dir, test_file), 'r') as f:
             diffs = get_diffs(f.readlines(), result.splitlines(keepends=True))
 
-            self.assertEqual(1, len(diffs['additions']), f'should have 1 addition, has these: {diffs["additions"]}')
-            self.assertEqual(0, len(diffs['deletions']), f'should have 0 deletions, has these: {diffs["deletions"]}')
+            assert 1 == len(diffs['additions']), f'should have 1 addition, has these: {diffs["additions"]}'
```
Comment (nllong) [inline tests/test_builder.py:106]: 💔

File: modelica_builder/model.py [hunk 3]
```diff
@@ -307,5 +307,8 @@ def save_as(self, filename):
         :param filename: string, name of file
         """
         result = self.execute()
-        with open(filename, 'w') as f:
+        # Setting newline=None here means any \n we added will be translated
+        # to the native OS's line separator:
+        # https://docs.python.org/3.6/library/functions.html#open
+        with open(filename, 'wt', newline=None) as f:
```
Comment (nllong) [inline modelica_builder/model.py:313]: nice fix! 🔥

**Review by nllong — APPROVED** [review]
tests passed, looks amazing!

File: tests/test_builder.py (hunk 2 above)
Comment (macintoshpie) [reply tests/test_builder.py:106]: 😿

**Pull request merged**